packages/idempotency/src/middleware/makeHandlerIdempotent.ts

```diff
@@ -0,0 +1,144 @@
+import { IdempotencyHandler } from '../IdempotencyHandler';
+import { IdempotencyConfig } from '../IdempotencyConfig';
+import { cleanupMiddlewares } from '@aws-lambda-powertools/commons/lib/middleware';
+import {
+  IdempotencyItemAlreadyExistsError,
+  IdempotencyPersistenceLayerError,
+} from '../Exceptions';
+import { IdempotencyRecord } from '../persistence';
+import type {
+  MiddlewareLikeObj,
+  MiddyLikeRequest,
+} from '@aws-lambda-powertools/commons';
+import type { IdempotencyLambdaHandlerOptions } from '../types';
+
+/**
+ * A middy middleware to make your Lambda Handler idempotent.
+ *
+ * @example
+ * ```typescript
+ * import {
+ *   makeHandlerIdempotent,
+ *   DynamoDBPersistenceLayer,
+ * } from '@aws-lambda-powertools/idempotency';
+ * import middy from '@middy/core';
+ *
+ * const dynamoDBPersistenceLayer = new DynamoDBPersistenceLayer({
+ *   tableName: 'idempotencyTable',
+ * })
+ *
+ * const lambdaHandler = async (_event: unknown, _context: unknown) => {
+ *   //...
+ * };
+ *
+ * export const handler = middy(lambdaHandler)
+ *  .use(makeHandlerIdempotent({ persistenceStore: dynamoDBPersistenceLayer }));
+ * ```
+ *
+ * @param options - Options for the idempotency middleware
+ */
+const makeHandlerIdempotent = (
+  options: IdempotencyLambdaHandlerOptions
+): MiddlewareLikeObj => {
+  const idempotencyConfig = options.config
```

Comment:
Is it okay to instantiate this once and reuse them throughout different requests?

Asking because in the decorator and function wrapper we create a new one for each request.

Reply:
Confirmed that this is fine.